Extensions/Lighting/lightruntimeobject.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * @typedef {Object} LightObjectDataType
+ * @property {Object} content The base parameters of light object.
+ * @property {number} content.radius The radius of light object.
+ * @property {string} content.color A string representing color in hexadecimal format.
+ * @property {string} content.texture A string representing the name of texture used for light object.
+ * @property {boolean} content.debugMode true if the light objects shows debug graphics, false otherwise.
+ *
+ * @typedef {ObjectData & LightObjectDataType} LightObjectData
+ */
+
+/**
+ * Displays a Light object.
+ * @memberof gdjs
+ * @class LightRuntimeObject
+ * @extends RuntimeObject
+ * @param {gdjs.RuntimeScene} runtimeScene
+ * @param {LightObjectData} lightObjectData
+ */
+gdjs.LightRuntimeObject = function (runtimeScene, lightObjectData) {
+  gdjs.RuntimeObject.call(this, runtimeScene, lightObjectData);
+
+  /** @type {number} */
+  this._radius =
+    lightObjectData.content.radius > 0 ? lightObjectData.content.radius : 1;
+
+  /** @type {number[]} color in format [r, g, b], where each component is in the range [0, 255] */
+  this._color = gdjs.LightRuntimeObject.hexToRGBColor(
+    lightObjectData.content.color
+  );
+
+  /** @type {boolean} */
+  this._debugMode = lightObjectData.content.debugMode;
+
+  /** @type {?PIXI.Texture} */
+  this._texture =
+    lightObjectData.content.texture === ''
+      ? null
+      : runtimeScene
+          .getGame()
+          .getImageManager()
+          .getPIXITexture(lightObjectData.content.texture);
```

Comment:
I unfortunately missed this, but the PIXI Texture should in theory be hold only in the "pixi renderer". Here, we should just store something that is "renderer-agnostic", i.e: the texture resource name (a string).

Could you do a PR to make `_texture` a string? Use `getPIXITexture` in the renderer. If you know that the texture was changed, then call a method called `updateTexture` of the renderer - which will do the job of updating the PIXI.Texture.

This way, we have an object that could work with something entirely different than PIXI in the future :)

Reply:
(Actually updateTexture already exists, and can be updated :))